src/server/user-controller.spec.js

```diff
@@ -219,6 +222,119 @@ describe('user-controller', () => {
 
   });
 
+  describe('handlePostUserRegistrationCompletion', () => {
+    let req;
+    let res;
+    const dbUser = { _id: 'cnztc5ztm41mx03z' };
+    const mappedUser = { _id: 'cnztc5ztm41mx03z' };
+
+    describe('when a pending user with the specified verification code exists', () => {
+      describe('and there is no external account in the session', () => {
+        beforeEach(() => new Promise((resolve, reject) => {
+          req = httpMocks.createRequest({
+            protocol: 'https',
+            headers: { host: 'localhost' },
+            session: { externalAccount: null },
+            body: { userId: 'cnztc5ztm41mx03z', verificationCode: '3xzrxzt43z0xtm1' },
+            login: sandbox.stub().callsArg(1)
+          });
+          res = httpMocks.createResponse({ eventEmitter: events.EventEmitter });
+
+          res.on('end', resolve);
+
+          userService.verifyUser.resolves(dbUser);
```

Comment:
Just a tiny thing, but for extra safety, we could set this up as `userService.verifyUser.withArgs(req.body.userId, req.body.verificationCode).resolves(dbUser)`